### PR TITLE
Reuses `mapModule` adding a map optional param

### PR DESCRIPTION
**Pull request by jbalsas**

### CODE REVIEW

**Comment by izaera** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ipeychev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: src/js/config-parser.js [hunk 1]
```diff
@@ -92,15 +92,30 @@ ConfigParser.prototype = {
...
      * @return {array|string} The mapped module or array of mapped modules.
      */
-    mapModule: function(module) {
-        if (!this._config.maps) {
+    mapModule: function(module, contextMap) {
+        if (!this._config.maps && !contextMap) {
             return module;
         }
 
+        var maps = {};
```
Comment (ipeychev) [inline src/js/config-parser.js]: Here we could do it with `Object.create(null);`

**Comment by ipeychev** [comment]
The rest of the source formatting issues I mentioned disappeared after the last commit 👍

**Comment by jbalsas** [comment]
Thanks @ipeychev!

@izaera is double checking this and contrasting it with the expectations of https://github.com/liferay/liferay-amd-loader/pull/127 to see what needs to be adapted. 

I'll apply that change if we go ahead with this :)

**Comment by izaera** [comment]
This PR can be closed (I don't have permissions :-( ) as it's been superseded by https://github.com/liferay/liferay-amd-loader/issues/131

**Pull request closed without merging**